Break in code-gen for external types used in path/query params

## What happened?
A panic was encountered after upgrading from `v6.5.0` to `v.6.6.0`, which includes the code-gen change to jennifer from goastwriter.
Of note, this appears when an API uses an external type in either path or query parameters.

```
panic: unrecognized type BackupId (string)

goroutine 1 [running]:
github.com/palantir/conjure-go/v6/conjure.astForDecodeHTTPParamInternal(0xc00038fe60, {0xc00023a588, 0x8}, {0x1465920, 0xc00041e080}, {0xc00023a588, 0x8}, {0x145a6f0, 0xc000389a58}, 0x0)
        /Volumes/git/go/src/github.com/palantir/conjure-go/conjure/serverwriter.go:387 +0x980d
github.com/palantir/conjure-go/v6/conjure.astForDecodeHTTPParam(...)
        /Volumes/git/go/src/github.com/palantir/conjure-go/conjure/serverwriter.go:289
github.com/palantir/conjure-go/v6/conjure.astForHandlerMethodQueryParam(0x0, 0xc000028d80)
        /Volumes/git/go/src/github.com/palantir/conjure-go/conjure/serverwriter.go:246 +0x1247
github.com/palantir/conjure-go/v6/conjure.astForHandlerMethodQueryParams(0x0, {0xc00034e378, 0x1, 0x0})
        /Volumes/git/go/src/github.com/palantir/conjure-go/conjure/serverwriter.go:234 +0x38
github.com/palantir/conjure-go/v6/conjure.astForHandlerMethodBody(0x100e367, {0xc0004e9020, 0xb}, 0xc0003a4bb0)
        /Volumes/git/go/src/github.com/palantir/conjure-go/conjure/serverwriter.go:149 +0x115
github.com/palantir/conjure-go/v6/conjure.astForHandlerMethods.func1(0xc00005ad80)
        /Volumes/git/go/src/github.com/palantir/conjure-go/conjure/serverwriter.go:137 +0x2c
github.com/dave/jennifer/jen.(*Statement).BlockFunc(0xc0002beee8, 0xc0005cb6a8)
        /Volumes/git/go/src/github.com/palantir/conjure-go/vendor/github.com/dave/jennifer/jen/generated.go:234 +0x89
github.com/palantir/conjure-go/v6/conjure.astForHandlerMethods(0xc00033e1e0)
        /Volumes/git/go/src/github.com/palantir/conjure-go/conjure/serverwriter.go:136 +0x1265
github.com/palantir/conjure-go/v6/conjure.writeServerType(0xc000285980, 0xc00033e1e0)
        /Volumes/git/go/src/github.com/palantir/conjure-go/conjure/serverwriter.go:54 +0x597
github.com/palantir/conjure-go/v6/conjure.GenerateOutputFiles({0x1, {0x1edea50, 0x0, 0x0}, {0xc00019a540, 0x4, 0x4}, {0xc0001f4400, 0x1, 0x4}, ...}, ...)
        /Volumes/git/go/src/github.com/palantir/conjure-go/conjure/conjure.go:95 +0x1005
github.com/palantir/conjure-go/v6/conjure.Generate({0x1, {0x1edea50, 0x0, 0x0}, {0xc00019a540, 0x4, 0x4}, {0xc0001f4400, 0x1, 0x4}, ...}, ...)
        /Volumes/git/go/src/github.com/palantir/conjure-go/conjure/conjure.go:29 +0x65
main.run({0x13d4413, 0x480}, {0x13b2b14, 0x6})
        /Volumes/git/go/src/github.com/palantir/conjure-go/integration_test/testgenerated/generate.go:61 +0x14f
main.main()
        /Volumes/git/go/src/github.com/palantir/conjure-go/integration_test/testgenerated/generate.go:33 +0xa5
exit status 2
```

## What did you want to happen?
Code should be generated as it was prior to `v6.6.0`

--- a/changelog/@unreleased/pr-248.v2.yml
+++ b/changelog/@unreleased/pr-248.v2.yml
@@ -1,0 +1,7 @@
+type: fix
+fix:
+  description: |-
+    Fix code generation of request param parsing for external types.
+    Fixes: #247
+  links:
+  - https://github.com/palantir/conjure-go/pull/248

--- a/conjure/serverwriter.go
+++ b/conjure/serverwriter.go
@@ -383,6 +383,8 @@ func astForDecodeHTTPParamInternal(methodBody *jen.Group, argName string, argTyp
 		)
 	case *types.Map, *types.ObjectType, *types.UnionType:
 		panic(fmt.Sprintf("unsupported complex type for http param %v", argType))
+	case *types.External:
+		astForDecodeHTTPParamInternal(methodBody, argName, typVal.Fallback, outVarName, inStrExpr, depth+1)
 	default:
 		panic(fmt.Sprintf("unrecognized type %v", argType))
 	}

--- a/conjure/serverwriter_test.go
+++ b/conjure/serverwriter_test.go
@@ -19,6 +19,7 @@ import (
 	"testing"
 
 	"github.com/dave/jennifer/jen"
+	"github.com/palantir/conjure-go/v6/conjure-api/conjure/spec"
 	"github.com/palantir/conjure-go/v6/conjure/types"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
@@ -428,6 +429,92 @@ func TestServerASTDecodeHTTPParam(t *testing.T) {
 	var myParam MyEnum
 	if err := myParam.UnmarshalText([]byte(req.URL.Query().Get("myParam"))); err != nil {
 		return witchcraftgoerror.WrapWithContextParams(req.Context(), errors.WrapWithInvalidArgument(err), "failed to parse \"myParam\" as MyEnum")
+	}
+}`,
+		},
+		{
+			Name: "external string query param",
+			Arg: types.EndpointArgumentDefinition{
+				Name: "myParam",
+				Type: &types.External{
+					Spec: spec.TypeName{
+						Name:    "myParam",
+						Package: "com.palantir.service.api.MyParam",
+					},
+					Fallback: types.String{},
+				},
+				ParamType: types.QueryParam,
+				ParamID:   "myParam",
+			},
+			Out: `{
+	myParam := req.URL.Query().Get("myParam")
+}`,
+		},
+		{
+			Name: "external string path param",
+			Arg: types.EndpointArgumentDefinition{
+				Name: "myParam",
+				Type: &types.External{
+					Spec: spec.TypeName{
+						Name:    "myParam",
+						Package: "com.palantir.service.api.MyParam",
+					},
+					Fallback: types.String{},
+				},
+				ParamType: types.PathParam,
+				ParamID:   "myParam",
+			},
+			Out: `{
+	myParamStr, ok := pathParams["myParam"]
+	if !ok {
+		return witchcraftgoerror.WrapWithContextParams(req.Context(), errors.NewInvalidArgument(), "path parameter \"myParam\" not present")
+	}
+	myParam := myParamStr
+}`,
+		},
+		{
+			Name: "External integer query param",
+			Arg: types.EndpointArgumentDefinition{
+				Name: "myParam",
+				Type: &types.External{
+					Spec: spec.TypeName{
+						Name:    "myParam",
+						Package: "com.palantir.service.api.MyParam",
+					},
+					Fallback: types.Integer{},
+				},
+				ParamType: types.QueryParam,
+				ParamID:   "myParam",
+			},
+			Out: `{
+	myParam, err := strconv.Atoi(req.URL.Query().Get("myParam"))
+	if err != nil {
+		return witchcraftgoerror.WrapWithContextParams(req.Context(), errors.WrapWithInvalidArgument(err), "failed to parse \"myParam\" as integer")
+	}
+}`,
+		},
+		{
+			Name: "External integer path param",
+			Arg: types.EndpointArgumentDefinition{
+				Name: "myParam",
+				Type: &types.External{
+					Spec: spec.TypeName{
+						Name:    "myParam",
+						Package: "com.palantir.service.api.MyParam",
+					},
+					Fallback: types.Integer{},
+				},
+				ParamType: types.PathParam,
+				ParamID:   "myParam",
+			},
+			Out: `{
+	myParamStr, ok := pathParams["myParam"]
+	if !ok {
+		return witchcraftgoerror.WrapWithContextParams(req.Context(), errors.NewInvalidArgument(), "path parameter \"myParam\" not present")
+	}
+	myParam, err := strconv.Atoi(myParamStr)
+	if err != nil {
+		return witchcraftgoerror.WrapWithContextParams(req.Context(), errors.WrapWithInvalidArgument(err), "failed to parse \"myParam\" as integer")
 	}
 }`,
 		},

--- a/integration_test/testgenerated/server/api/servers.conjure.go
+++ b/integration_test/testgenerated/server/api/servers.conjure.go
@@ -39,6 +39,10 @@ type TestService interface {
 	QueryParamListSafeLong(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg []safelong.SafeLong) error
 	QueryParamListString(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg []string) error
 	QueryParamListUuid(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg []uuid.UUID) error
+	QueryParamExternalString(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg string) error
+	QueryParamExternalInteger(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg int) error
+	PathParamExternalString(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg string) error
+	PathParamExternalInteger(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg int) error
 	PostPathParam(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg string, myPathParam2Arg bool, myBodyParamArg CustomObject, myQueryParam1Arg string, myQueryParam2Arg string, myQueryParam3Arg float64, myQueryParam4Arg *safelong.SafeLong, myQueryParam5Arg *string, myQueryParam6Arg OptionalIntegerAlias, myHeaderParam1Arg safelong.SafeLong, myHeaderParam2Arg *uuid.UUID) (CustomObject, error)
 	PostSafeParams(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg string, myPathParam2Arg bool, myBodyParamArg CustomObject, myQueryParam1Arg string, myQueryParam2Arg string, myQueryParam3Arg float64, myQueryParam4Arg *safelong.SafeLong, myQueryParam5Arg *string, myHeaderParam1Arg safelong.SafeLong, myHeaderParam2Arg *uuid.UUID) error
 	Bytes(ctx context.Context) (CustomObject, error)
@@ -104,6 +108,18 @@ func RegisterRoutesTestService(router wrouter.Router, impl TestService) error {
 	}
 	if err := resource.Get("QueryParamListUuid", "/uuidListQueryVar", httpserver.NewJSONHandler(handler.HandleQueryParamListUuid, httpserver.StatusCodeMapper, httpserver.ErrHandler)); err != nil {
 		return werror.Wrap(err, "failed to add queryParamListUuid route")
+	}
+	if err := resource.Get("QueryParamExternalString", "/externalStringQueryVar", httpserver.NewJSONHandler(handler.HandleQueryParamExternalString, httpserver.StatusCodeMapper, httpserver.ErrHandler)); err != nil {
+		return werror.Wrap(err, "failed to add queryParamExternalString route")
+	}
+	if err := resource.Get("QueryParamExternalInteger", "/externalIntegerQueryVar", httpserver.NewJSONHandler(handler.HandleQueryParamExternalInteger, httpserver.StatusCodeMapper, httpserver.ErrHandler)); err != nil {
+		return werror.Wrap(err, "failed to add queryParamExternalInteger route")
+	}
+	if err := resource.Post("PathParamExternalString", "/externalStringPath/{myPathParam1}", httpserver.NewJSONHandler(handler.HandlePathParamExternalString, httpserver.StatusCodeMapper, httpserver.ErrHandler)); err != nil {
+		return werror.Wrap(err, "failed to add pathParamExternalString route")
+	}
+	if err := resource.Post("PathParamExternalInteger", "/externalIntegerPath/{myPathParam1}", httpserver.NewJSONHandler(handler.HandlePathParamExternalInteger, httpserver.StatusCodeMapper, httpserver.ErrHandler)); err != nil {
+		return werror.Wrap(err, "failed to add pathParamExternalInteger route")
 	}
 	if err := resource.Post("PostPathParam", "/path/{myPathParam1}/{myPathParam2}", httpserver.NewJSONHandler(handler.HandlePostPathParam, httpserver.StatusCodeMapper, httpserver.ErrHandler)); err != nil {
 		return werror.Wrap(err, "failed to add postPathParam route")
@@ -423,6 +439,80 @@ func (t *testServiceHandler) HandleQueryParamListUuid(rw http.ResponseWriter, re
 		myQueryParam1 = append(myQueryParam1, convertedVal)
 	}
 	if err := t.impl.QueryParamListUuid(req.Context(), bearertoken.Token(authHeader), myQueryParam1); err != nil {
+		return err
+	}
+	rw.WriteHeader(http.StatusNoContent)
+	return nil
+}
+
+func (t *testServiceHandler) HandleQueryParamExternalString(rw http.ResponseWriter, req *http.Request) error {
+	authHeader, err := httpserver.ParseBearerTokenHeader(req)
+	if err != nil {
+		return errors.WrapWithPermissionDenied(err)
+	}
+	myQueryParam1 := req.URL.Query().Get("myQueryParam1")
+	if err := t.impl.QueryParamExternalString(req.Context(), bearertoken.Token(authHeader), myQueryParam1); err != nil {
+		return err
+	}
+	rw.WriteHeader(http.StatusNoContent)
+	return nil
+}
+
+func (t *testServiceHandler) HandleQueryParamExternalInteger(rw http.ResponseWriter, req *http.Request) error {
+	authHeader, err := httpserver.ParseBearerTokenHeader(req)
+	if err != nil {
+		return errors.WrapWithPermissionDenied(err)
+	}
+	myQueryParam1, err := strconv.Atoi(req.URL.Query().Get("myQueryParam1"))
+	if err != nil {
+		return werror.WrapWithContextParams(req.Context(), errors.WrapWithInvalidArgument(err), "failed to parse \"myQueryParam1\" as integer")
+	}
+	if err := t.impl.QueryParamExternalInteger(req.Context(), bearertoken.Token(authHeader), myQueryParam1); err != nil {
+		return err
+	}
+	rw.WriteHeader(http.StatusNoContent)
+	return nil
+}
+
+func (t *testServiceHandler) HandlePathParamExternalString(rw http.ResponseWriter, req *http.Request) error {
+	authHeader, err := httpserver.ParseBearerTokenHeader(req)
+	if err != nil {
+		return errors.WrapWithPermissionDenied(err)
+	}
+	pathParams := wrouter.PathParams(req)
+	if pathParams == nil {
+		return werror.Wrap(errors.NewInternal(), "path params not found on request: ensure this endpoint is registered with wrouter")
+	}
+	myPathParam1Str, ok := pathParams["myPathParam1"]
+	if !ok {
+		return werror.WrapWithContextParams(req.Context(), errors.NewInvalidArgument(), "path parameter \"myPathParam1\" not present")
+	}
+	myPathParam1 := myPathParam1Str
+	if err := t.impl.PathParamExternalString(req.Context(), bearertoken.Token(authHeader), myPathParam1); err != nil {
+		return err
+	}
+	rw.WriteHeader(http.StatusNoContent)
+	return nil
+}
+
+func (t *testServiceHandler) HandlePathParamExternalInteger(rw http.ResponseWriter, req *http.Request) error {
+	authHeader, err := httpserver.ParseBearerTokenHeader(req)
+	if err != nil {
+		return errors.WrapWithPermissionDenied(err)
+	}
+	pathParams := wrouter.PathParams(req)
+	if pathParams == nil {
+		return werror.Wrap(errors.NewInternal(), "path params not found on request: ensure this endpoint is registered with wrouter")
+	}
+	myPathParam1Str, ok := pathParams["myPathParam1"]
+	if !ok {
+		return werror.WrapWithContextParams(req.Context(), errors.NewInvalidArgument(), "path parameter \"myPathParam1\" not present")
+	}
+	myPathParam1, err := strconv.Atoi(myPathParam1Str)
+	if err != nil {
+		return werror.WrapWithContextParams(req.Context(), errors.WrapWithInvalidArgument(err), "failed to parse \"myPathParam1\" as integer")
+	}
+	if err := t.impl.PathParamExternalInteger(req.Context(), bearertoken.Token(authHeader), myPathParam1); err != nil {
 		return err
 	}
 	rw.WriteHeader(http.StatusNoContent)

--- a/integration_test/testgenerated/server/api/services.conjure.go
+++ b/integration_test/testgenerated/server/api/services.conjure.go
@@ -35,6 +35,10 @@ type TestServiceClient interface {
 	QueryParamListSafeLong(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg []safelong.SafeLong) error
 	QueryParamListString(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg []string) error
 	QueryParamListUuid(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg []uuid.UUID) error
+	QueryParamExternalString(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg string) error
+	QueryParamExternalInteger(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg int) error
+	PathParamExternalString(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg string) error
+	PathParamExternalInteger(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg int) error
 	PostPathParam(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg string, myPathParam2Arg bool, myBodyParamArg CustomObject, myQueryParam1Arg string, myQueryParam2Arg string, myQueryParam3Arg float64, myQueryParam4Arg *safelong.SafeLong, myQueryParam5Arg *string, myQueryParam6Arg OptionalIntegerAlias, myHeaderParam1Arg safelong.SafeLong, myHeaderParam2Arg *uuid.UUID) (CustomObject, error)
 	PostSafeParams(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg string, myPathParam2Arg bool, myBodyParamArg CustomObject, myQueryParam1Arg string, myQueryParam2Arg string, myQueryParam3Arg float64, myQueryParam4Arg *safelong.SafeLong, myQueryParam5Arg *string, myHeaderParam1Arg safelong.SafeLong, myHeaderParam2Arg *uuid.UUID) error
 	Bytes(ctx context.Context) (CustomObject, error)
@@ -310,6 +314,60 @@ func (c *testServiceClient) QueryParamListUuid(ctx context.Context, authHeader b
 	return nil
 }
 
+func (c *testServiceClient) QueryParamExternalString(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg string) error {
+	var requestParams []httpclient.RequestParam
+	requestParams = append(requestParams, httpclient.WithRPCMethodName("QueryParamExternalString"))
+	requestParams = append(requestParams, httpclient.WithRequestMethod("GET"))
+	requestParams = append(requestParams, httpclient.WithHeader("Authorization", fmt.Sprint("Bearer ", authHeader)))
+	requestParams = append(requestParams, httpclient.WithPathf("/externalStringQueryVar"))
+	queryParams := make(url.Values)
+	queryParams.Set("myQueryParam1", fmt.Sprint(myQueryParam1Arg))
+	requestParams = append(requestParams, httpclient.WithQueryValues(queryParams))
+	if _, err := c.client.Do(ctx, requestParams...); err != nil {
+		return werror.WrapWithContextParams(ctx, err, "queryParamExternalString failed")
+	}
+	return nil
+}
+
+func (c *testServiceClient) QueryParamExternalInteger(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg int) error {
+	var requestParams []httpclient.RequestParam
+	requestParams = append(requestParams, httpclient.WithRPCMethodName("QueryParamExternalInteger"))
+	requestParams = append(requestParams, httpclient.WithRequestMethod("GET"))
+	requestParams = append(requestParams, httpclient.WithHeader("Authorization", fmt.Sprint("Bearer ", authHeader)))
+	requestParams = append(requestParams, httpclient.WithPathf("/externalIntegerQueryVar"))
+	queryParams := make(url.Values)
+	queryParams.Set("myQueryParam1", fmt.Sprint(myQueryParam1Arg))
+	requestParams = append(requestParams, httpclient.WithQueryValues(queryParams))
+	if _, err := c.client.Do(ctx, requestParams...); err != nil {
+		return werror.WrapWithContextParams(ctx, err, "queryParamExternalInteger failed")
+	}
+	return nil
+}
+
+func (c *testServiceClient) PathParamExternalString(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg string) error {
+	var requestParams []httpclient.RequestParam
+	requestParams = append(requestParams, httpclient.WithRPCMethodName("PathParamExternalString"))
+	requestParams = append(requestParams, httpclient.WithRequestMethod("POST"))
+	requestParams = append(requestParams, httpclient.WithHeader("Authorization", fmt.Sprint("Bearer ", authHeader)))
+	requestParams = append(requestParams, httpclient.WithPathf("/externalStringPath/%s", url.PathEscape(fmt.Sprint(myPathParam1Arg))))
+	if _, err := c.client.Do(ctx, requestParams...); err != nil {
+		return werror.WrapWithContextParams(ctx, err, "pathParamExternalString failed")
+	}
+	return nil
+}
+
+func (c *testServiceClient) PathParamExternalInteger(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg int) error {
+	var requestParams []httpclient.RequestParam
+	requestParams = append(requestParams, httpclient.WithRPCMethodName("PathParamExternalInteger"))
+	requestParams = append(requestParams, httpclient.WithRequestMethod("POST"))
+	requestParams = append(requestParams, httpclient.WithHeader("Authorization", fmt.Sprint("Bearer ", authHeader)))
+	requestParams = append(requestParams, httpclient.WithPathf("/externalIntegerPath/%s", url.PathEscape(fmt.Sprint(myPathParam1Arg))))
+	if _, err := c.client.Do(ctx, requestParams...); err != nil {
+		return werror.WrapWithContextParams(ctx, err, "pathParamExternalInteger failed")
+	}
+	return nil
+}
+
 func (c *testServiceClient) PostPathParam(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg string, myPathParam2Arg bool, myBodyParamArg CustomObject, myQueryParam1Arg string, myQueryParam2Arg string, myQueryParam3Arg float64, myQueryParam4Arg *safelong.SafeLong, myQueryParam5Arg *string, myQueryParam6Arg OptionalIntegerAlias, myHeaderParam1Arg safelong.SafeLong, myHeaderParam2Arg *uuid.UUID) (CustomObject, error) {
 	var defaultReturnVal CustomObject
 	var returnVal *CustomObject
@@ -480,6 +538,10 @@ type TestServiceClientWithAuth interface {
 	QueryParamListSafeLong(ctx context.Context, myQueryParam1Arg []safelong.SafeLong) error
 	QueryParamListString(ctx context.Context, myQueryParam1Arg []string) error
 	QueryParamListUuid(ctx context.Context, myQueryParam1Arg []uuid.UUID) error
+	QueryParamExternalString(ctx context.Context, myQueryParam1Arg string) error
+	QueryParamExternalInteger(ctx context.Context, myQueryParam1Arg int) error
+	PathParamExternalString(ctx context.Context, myPathParam1Arg string) error
+	PathParamExternalInteger(ctx context.Context, myPathParam1Arg int) error
 	PostPathParam(ctx context.Context, myPathParam1Arg string, myPathParam2Arg bool, myBodyParamArg CustomObject, myQueryParam1Arg string, myQueryParam2Arg string, myQueryParam3Arg float64, myQueryParam4Arg *safelong.SafeLong, myQueryParam5Arg *string, myQueryParam6Arg OptionalIntegerAlias, myHeaderParam1Arg safelong.SafeLong, myHeaderParam2Arg *uuid.UUID) (CustomObject, error)
 	PostSafeParams(ctx context.Context, myPathParam1Arg string, myPathParam2Arg bool, myBodyParamArg CustomObject, myQueryParam1Arg string, myQueryParam2Arg string, myQueryParam3Arg float64, myQueryParam4Arg *safelong.SafeLong, myQueryParam5Arg *string, myHeaderParam1Arg safelong.SafeLong, myHeaderParam2Arg *uuid.UUID) error
 	Bytes(ctx context.Context) (CustomObject, error)
@@ -563,6 +625,22 @@ func (c *testServiceClientWithAuth) QueryParamListString(ctx context.Context, my
 
 func (c *testServiceClientWithAuth) QueryParamListUuid(ctx context.Context, myQueryParam1Arg []uuid.UUID) error {
 	return c.client.QueryParamListUuid(ctx, c.authHeader, myQueryParam1Arg)
+}
+
+func (c *testServiceClientWithAuth) QueryParamExternalString(ctx context.Context, myQueryParam1Arg string) error {
+	return c.client.QueryParamExternalString(ctx, c.authHeader, myQueryParam1Arg)
+}
+
+func (c *testServiceClientWithAuth) QueryParamExternalInteger(ctx context.Context, myQueryParam1Arg int) error {
+	return c.client.QueryParamExternalInteger(ctx, c.authHeader, myQueryParam1Arg)
+}
+
+func (c *testServiceClientWithAuth) PathParamExternalString(ctx context.Context, myPathParam1Arg string) error {
+	return c.client.PathParamExternalString(ctx, c.authHeader, myPathParam1Arg)
+}
+
+func (c *testServiceClientWithAuth) PathParamExternalInteger(ctx context.Context, myPathParam1Arg int) error {
+	return c.client.PathParamExternalInteger(ctx, c.authHeader, myPathParam1Arg)
 }
 
 func (c *testServiceClientWithAuth) PostPathParam(ctx context.Context, myPathParam1Arg string, myPathParam2Arg bool, myBodyParamArg CustomObject, myQueryParam1Arg string, myQueryParam2Arg string, myQueryParam3Arg float64, myQueryParam4Arg *safelong.SafeLong, myQueryParam5Arg *string, myQueryParam6Arg OptionalIntegerAlias, myHeaderParam1Arg safelong.SafeLong, myHeaderParam2Arg *uuid.UUID) (CustomObject, error) {

--- a/integration_test/testgenerated/server/server-service.yml
+++ b/integration_test/testgenerated/server/server-service.yml
@@ -6,6 +6,14 @@ types:
     OtherMarker:
       external:
         java: com.palantir.logsafe.OtherMarker
+    ExternalString:
+      base-type: string
+      external:
+        java: com.palantir.service.api.ExternalString
+    ExternalInteger:
+      base-type: integer
+      external:
+        java: com.palantir.rescue.api.ExternalInteger
   definitions:
     default-package: api
     objects:
@@ -119,6 +127,32 @@ services:
           myQueryParam1:
             type: list<uuid>
             param-type: query
+      queryParamExternalString:
+        http: GET /externalStringQueryVar
+        auth: header
+        args:
+          myQueryParam1:
+            type: ExternalString
+            param-type: query
+      queryParamExternalInteger:
+        http: GET /externalIntegerQueryVar
+        auth: header
+        args:
+          myQueryParam1:
+            type: ExternalInteger
+            param-type: query
+      pathParamExternalString:
+        http: POST /externalStringPath/{myPathParam1}
+        auth: header
+        args:
+          myPathParam1:
+            type: ExternalString
+      pathParamExternalInteger:
+        http: POST /externalIntegerPath/{myPathParam1}
+        auth: header
+        args:
+          myPathParam1:
+            type: ExternalInteger
       postPathParam:
         http: POST /path/{myPathParam1}/{myPathParam2}
         auth: header

--- a/integration_test/testgenerated/server/server_test.go
+++ b/integration_test/testgenerated/server/server_test.go
@@ -363,3 +363,19 @@ func (t testServerImpl) GetOptionalBinary(ctx context.Context) (*io.ReadCloser, 
 func (t testServerImpl) Chan(ctx context.Context, varArg string, importArg map[string]string, typeArg string, returnArg safelong.SafeLong) error {
 	panic("implement me")
 }
+
+func (V testServerImpl) QueryParamExternalString(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg string) error {
+	panic("implement me")
+}
+
+func (V testServerImpl) QueryParamExternalInteger(ctx context.Context, authHeader bearertoken.Token, myQueryParam1Arg int) error {
+	panic("implement me")
+}
+
+func (V testServerImpl) PathParamExternalString(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg string) error {
+	panic("implement me")
+}
+
+func (V testServerImpl) PathParamExternalInteger(ctx context.Context, authHeader bearertoken.Token, myPathParam1Arg int) error {
+	panic("implement me")
+}

